orion/sites/admin/src/location/location.js

```diff
@@ -20,29 +24,97 @@ export class LocationPage extends React.Component {
         });
     }
 
+    onCheckRow() {
+        if (document.getElementById("unpublished").checked) {
+            this.setState({ numSelected: this.state.numSelected + 1 });
+        } else {
+            this.setState({ numSelected: this.state.numSelected - 1 });
```

Comment:
And oh yea, if you're going to loop, make sure to call this.setState() once AFTER the loop finishes.
this.setState() is actually an asynchronous call meaning it will take a while (several milliseconds) until React actually updates the state variables to then rerender data.

So rather than have 5 asynchronous calls to increment some variable by 1 at a time.
It's better to call one asynchronous call, and simply increment by 5.
Hope that makes sense.